File: Haskell/PL_HW-2/HaskellCode/MaxofList.hs
--- find the maximum number in a given list -----

maxofList :: Integral a => [a] -> a 
maxofList [] = error "no values in the list"
maxofList [x] = x 
maxofList (x:xs) = helper xs x


helper :: Integral a => [a] -> a -> a 
helper [] currentmax = currentmax
helper (x:xs) currentmax = 
  if (x > currentmax)
   then helper xs x 
  else 
    helper xs currentmax   

main = do{
          print(maxofList [1,2,3,4,5,6,76,86]);
        }

File: Prolog/PL_HW-2/PrologCode/MaxofList.pl
% program to find the max value of the
%  elements of the given list

% fact 1, the terminal case 
% X is the max element of list with element X
maxoflist([X],X).

maxoflist([Start|EndTail],X) :-
    maxoflist(EndTail,Y), (Start > Y -> X=Start; Y=X).
% if else condition to check if the first element is greater than the currentMax   
   
   